packages/fxa-settings/src/pages/AccountRecoveryResetPassword/index.tsx

```diff
@@ -0,0 +1,233 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps } from '@reach/router';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from '@reach/router';
+import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
+import { useAccount, useAlertBar } from '../../models';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models/hooks';
+
+import { InputText } from '../../components/InputText';
+import LinkRememberPassword from '../../components/LinkRememberPassword';
+// --canGoBack-- determines if the user can navigate back to an fxa entrypoint
+
+export type AccountRecoveryResetPasswordProps = {
+  canGoBack?: boolean;
+  linkStatus: LinkStatus;
+};
+
+type FormData = {
+  currentPassword: string;
+  newPassword: string;
+};
+
+type LinkStatus = 'expired' | 'broken' | 'valid';
+
+const AccountRecoveryResetPassword = ({
+  canGoBack,
+  linkStatus,
+}: AccountRecoveryResetPasswordProps & RouteComponentProps) => {
+  usePageViewEvent('account-recovery-reset-password', {
+    entrypoint_variation: 'react',
+  });
+
+  const [currentPassword, setCurrentPassword] = useState<string>('');
+  const [newPassword, setNewPassword] = useState<string>('');
+  const [currentPasswordErrorText, setCurrentPasswordErrorText] =
+    useState<string>('');
+  const [newPasswordErrorText, setNewPasswordErrorText] = useState<string>('');
+  const [isFocused, setIsFocused] = useState(false);
+  const alertBar = useAlertBar();
+  const account = useAccount();
+  const navigate = useNavigate();
+  const onFocusMetricsEvent = 'account-recovery-reset-password.engage';
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const { handleSubmit } = useForm<FormData>({
+    mode: 'onBlur',
+    criteriaMode: 'all',
+    defaultValues: {
+      currentPassword: '',
+      newPassword: '',
+    },
+  });
+
+  const onFocus = () => {
+    if (!isFocused && onFocusMetricsEvent) {
+      logViewEvent('flow', onFocusMetricsEvent, {
+        entrypoint_variation: 'react',
+      });
+      setIsFocused(true);
+    }
+  };
+
+  const navigateToHome = useCallback(() => {
+    navigate('/settings', { replace: true });
+  }, [navigate]);
+
+  // TO-DO:
+  // * Set tooltip error message reflecting password requirements.
+  // const setErrorMessage = (errorText: string) => {
+  //   setEmailErrorText(errorText);
+  // };
+  // * Set up metrics for all events
+  // - submitting the new password
+  // - focusing the password inputs if desired
+  // - remembering one's password
+  // - asking for a new link.
+  // * Hook up the functionality for sending a user a new verification link,
+  // instead of this dummy function.
+  const sendNewLinkEmail = () => {};
+
+  const onSubmit = () => {
+    try {
+      account.changePassword(currentPassword, newPassword);
```

Comment:
The `changePassword` assumes you are logged in already, this is where we will need to add support for new gql mutations from https://github.com/mozilla/fxa/pull/14564.